gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -801,23 +801,25 @@ private void addJobConfPackage(String jobConfPackagePath, Path destDir, Map<Stri
 
   @VisibleForTesting
   protected String buildApplicationMasterCommand(String applicationId, int memoryMbs) {
-    String appMasterClassName = GobblinApplicationMaster.class.getSimpleName();
+    String appMasterClass = ConfigUtils.getString(
+       config, GobblinYarnConfigurationKeys.APP_MASTER_CLASS, GobblinYarnConfigurationKeys.DEFAULT_APP_MASTER_CLASS);
+    String logFileName = GobblinApplicationMaster.class.getSimpleName();
     return new StringBuilder()
         .append(ApplicationConstants.Environment.JAVA_HOME.$()).append("/bin/java")
         .append(" -Xmx").append((int) (memoryMbs * this.jvmMemoryXmxRatio) - this.jvmMemoryOverheadMbs).append("M")
         .append(" -D").append(GobblinYarnConfigurationKeys.JVM_USER_TIMEZONE_CONFIG).append("=").append(this.containerTimezone)
         .append(" -D").append(GobblinYarnConfigurationKeys.GOBBLIN_YARN_CONTAINER_LOG_DIR_NAME).append("=").append(ApplicationConstants.LOG_DIR_EXPANSION_VAR)
-        .append(" -D").append(GobblinYarnConfigurationKeys.GOBBLIN_YARN_CONTAINER_LOG_FILE_NAME).append("=").append(appMasterClassName).append(".").append(ApplicationConstants.STDOUT)
+        .append(" -D").append(GobblinYarnConfigurationKeys.GOBBLIN_YARN_CONTAINER_LOG_FILE_NAME).append("=").append(logFileName).append(".").append(ApplicationConstants.STDOUT)
         .append(" ").append(JvmUtils.formatJvmArguments(this.appMasterJvmArgs))
-        .append(" ").append(GobblinApplicationMaster.class.getName())
+        .append(" ").append(appMasterClass)
         .append(" --").append(GobblinClusterConfigurationKeys.APPLICATION_NAME_OPTION_NAME)
         .append(" ").append(this.applicationName)
         .append(" --").append(GobblinClusterConfigurationKeys.APPLICATION_ID_OPTION_NAME)
         .append(" ").append(applicationId)
         .append(" 1>").append(ApplicationConstants.LOG_DIR_EXPANSION_VAR).append(File.separator).append(
-            appMasterClassName).append(".").append(ApplicationConstants.STDOUT)
+            logFileName).append(".").append(ApplicationConstants.STDOUT)
```

Comment:
Why do we want to keep the log name as the GobblinApplicationMaster class name even we change the class? should we just keep it to be appMasterClass?

Reply:
With this originally being hard coded for so many years, I wasn't sure if it made sense to change this or not. I had some concerns about if any assumptions were made about the name of this class but for external systems.

I saw a few choices:
1. Change the log file name to always be the same as the class impl name
2. Make the log file name configurable to anything arbtirary, with the default being the log file name
3. Keep the log file always GobblinApplicationMaster.*

I didn't see much value in actually changing the name of the log files when changing the app master implementation class. To me, the first container launched by the `YarnAppLauncher` is always a yarn `ApplicationMaster`.  So it's never incorrect to refer to the log files produced by that container as a `GobblinApplicationMaster`. What are your thoughts? Do you think one of the other 2 choices makes more sense?

Reply:
I don't have a strong opinion here, just feel separately specify the log name to another class name seems wired to me. 
Also I did a code search and seems GobblinApplicationMaster is not referenced in any other places (retention, log agg etc)

Reply:
Good catch. In that case, I will have it match the class name by default with an option to override (that probably won't be used)

Reply:
I'm up for whatever... but personally I can't think of situations where I would choose to override the very sensible default:
```
Class.forName(appMasterClass).getSimpleName()
```